### ISSUE TITLE
functools.total_ordering obscures original type

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#443]
**Describe the bug**
I don't know what the semantics of `type` in function annotations are, so this might very well be an issue with the typeshed; but when decorating a class with `functools.total_ordering` the class type is lost, which leads pyright to report it's missing all of its original members:

```
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Cannot access member 'bar' for type 'type'\n  Member 'bar' is unknown",
	"source": "pyright",
	"startLineNumber": 9,
	"startColumn": 5,
	"endLineNumber": 9,
	"endColumn": 7
}
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That it should be able to locate the class members.

**Screenshots or Code**
```py
import functools


@functools.total_ordering
class Foo:
    bar = 1


Foo.bar  # This fails the check
```

**VS Code extension or command-line**
extension, 1.1.11

**Additional context**
-

**Comment by erictraut** [comment]
Thanks for filing the issue.

Yeah, the Python type system isn't expressive enough to describe the transform that functools.total_ordering provides. The typeshed definition of `total_ordering` is as follows:

```
def total_ordering(cls: type) -> type: ...
```

It indicates that the input of the decorator is a type (a class) and the output is a completely generic type. Based on my understanding of the `total_ordering` implementation, I would think that we could do better than this.

```
_T = TypeVar("_T")
def total_ordering(cls: _T) -> _T: ...
```

This would at least preserve all of the other methods in the class, even if it doesn't add the new comparison operators. 

I recommend opening an issue with the typeshed repo. Pyright consumes the typeshed stdllib stubs directly from that repo.

**Comment by layday** [comment]
Thanks a lot, I opened a PR there.
